src/tty_interface.c

```diff
@@ -118,13 +118,18 @@ static void action_emit(tty_interface_t *state) {
 	/* ttyout should be flushed before outputting on stdout */
 	tty_close(state->tty);
 
-	const char *selection = choices_get(state->choices, state->choices->selection);
-	if (selection) {
-		/* output the selected result */
-		printf("%s\n", selection);
-	} else {
-		/* No match, output the query instead */
-		printf("%s\n", state->search);
+	unsigned int all = state->options->output_all;
+	size_t start     = all ? 0 : state->choices->selection;
+	size_t end       = all ? choices_available(state->choices) : start + 1;
```

Comment:
It sure is prettier, but I think it's bad. Say you add a fourth statement with a longer type and variable name, now you have to update 4 lines of code. If you just always use ` = ` then a single line of code won't involve changing others, keeping git diffs more readable. Yep, bad idea.